test/eig.jl

```diff
@@ -16,16 +16,33 @@ using MatrixAlgebraKit: diagview
         @test eltype(D) == eltype(V) == Tc
         @test A * V ≈ V * D
 
+        alg′ = @constinferred MatrixAlgebraKit.select_algorithm(eig_full!, A, alg)
+
         Ac = similar(A)
-        D2, V2 = @constinferred eig_full!(copy!(Ac, A), (D, V), alg)
+        D2, V2 = @constinferred eig_full!(copy!(Ac, A), (D, V), alg′)
         @test D2 === D
         @test V2 === V
         @test A * V ≈ V * D
 
-        Dc = @constinferred eig_vals(A, alg)
+        Dc = @constinferred eig_vals(A, alg′)
         @test eltype(Dc) == Tc
         @test D ≈ Diagonal(Dc)
     end
+
+    # Test other alg inputs.
+    A = randn(rng, T, m, m)
+    Tc = complex(T)
+    D, V = @constinferred eig_full(A; alg=:LAPACK_Simple)
+    @test eltype(D) == eltype(V) == Tc
+    @test A * V ≈ V * D
+
+    A = randn(rng, T, m, m)
+    Tc = complex(T)
+    ## Inference is broken for this case for some reason.
+    ## D, V = @constinferred eig_full(A; alg=LAPACK_Simple)
```

Comment:
Note that this is still broken.

Reply:
This is fixed now, @lkdvos explained to me that I was using `@constinferred` wrong and should have been interpolating the `alg` input.